matchms/similarity/BaseSimilarity.py

```diff
@@ -59,3 +62,21 @@ def matrix(self, references: List[SpectrumType], queries: List[SpectrumType],
                 for i_query, query in enumerate(queries[:n_cols]):
                     scores[i_ref][i_query] = self.pair(reference, query)
         return scores
+
+    def sort(self, scores: numpy.ndarray):
+        """Return array of indexes for sorted list of scores.
+        This method can be adapted for different styles of scores.
+
+        Parameters
+        ----------
+        scores
+            1D Array of scores.
+
+        Returns
+        -------
+        idx_sorted
+            Indexes of sorted scores.
+        """
+        if scores.dtype.names is None:
+            return scores.argsort()[::-1]
+        return scores["score"].argsort()[::-1]
```

Comment:
Don't think the base class should look at the incoming dtype. It should sort according to the score_datatype. So it should only consist out of line 81. Line 82 is specific to the Cosine classes it should not be part of the base.

Classes which override score_datatype should have their own `sort()` if needed. The `sort() could be implemented in an intermediate abstract class or in the class itself.

Reply:
I reduced it to line 81 as suggested. In fact, that even works for all scores we implemented so far (without need to define own sort() method).